java/src/main/native/ai_onnxruntime_OrtTrainingSession.c

```diff
@@ -0,0 +1,745 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
```

Comment:
This would have been a lot less and safer code in C++

Reply:
Well it might have been if I'd written any C++ in the past 15 years, but as I haven't I'm not sure it would turn out much better.